apps/hash-frontend/src/pages/entities.page.tsx

```diff
@@ -88,16 +206,24 @@ const EntitiesPage: NextPageWithLayout = () => {
   });
 
   const isViewAllPagesPage =
-    entityType && isPageEntityTypeId(entityType.schema.$id);
+    entityType && entityType.schema.$id === systemEntityTypes.page.entityTypeId;
 
   const pageTitle = entityType
     ? entityTypeId
       ? `${entityType.schema.title} v${extractVersion(entityTypeId)}`
-      : `${entityType.schema.title}s`
+      : pluralize(entityType.schema.title)
     : "Entities";
 
   const { entities, loading } = entityTypeEntitiesValue;
 
+  const displayCreateEntityButton = useMemo(() => {
+    if (isViewAllPagesPage) {
+      return hashInstance?.properties.pagesAreEnabled;
+    }
+
+    return true;
+  }, [isViewAllPagesPage, hashInstance]);
```

Comment:
Sorry for not spotting this sooner, but we don't want to hide this button any more as far as I know – even if pages are disabled, we want the button for all the other entity types, including file. `pagesAreEnabled` should only affect the display of the page-related buttons.

Reply:
I'm not sure I understand, the reason I kept the logic in the parent component rather than just return `null` in the `CreateButtons` component is so that the buttons fade in if the pages page is visited and `hashInstance` is `undefined` on the first render, and on the second render `pagesAreEnabled` is evaluated to be `true`.

> even if pages are disabled, we want the button for all the other entity types, including file

Unless I'm missing something, I think that should be the case with the logic above.

Reply:
Sorry, I misread the logic, ignore this comment